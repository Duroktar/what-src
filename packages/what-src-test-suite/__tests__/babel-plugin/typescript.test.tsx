import * as babel from '@babel/core'
import { WhatSrcPluginOptions } from '@what-src/plugin-core'
import whatSrcBabelPlugin from '@what-src/babel-plugin'

const example = `
import * as React from 'react'

class TestClassComponent extends React.Component {
  render() {
    return (
      <div className="class-component">
        <p>some p thing</p>
      </div>
    )
  }
}

const TestFunctionalComponent = () => (
  <div className="functional-component">
    <p>some p thing</p>
  </div>
)

const TestMemoizedComponent = React.memo(() => (
  <div className="memoized-component">
    <p>some p thing</p>
  </div>
))
`

describe('works as a babel-plugin with preset-typescript', () => {
  it('works in development mode', () => {
    const res = babel.transformSync(example, {
      envName: 'development',
      plugins: [whatSrcBabelPlugin],
      filename: 'README.md',
      presets: [
        '@babel/preset-react',
        '@babel/preset-typescript',
      ],
    })

    expect(res).toHaveProperty('code')

    expect(res!.code).toMatchSnapshot()
  })

  it('works in production mode', () => {
    const configuration: WhatSrcPluginOptions = {
      useRemote: true,
      productionMode: true,
    }

    const res = babel.transformSync(example, {
      envName: 'production',
      plugins: [[whatSrcBabelPlugin, configuration]],
      filename: 'README.md',
      presets: [
        '@babel/preset-react',
        '@babel/preset-typescript',
      ],
    })

    expect(res).toHaveProperty('code')

    expect(res!.code).toMatchSnapshot()
  })
})
