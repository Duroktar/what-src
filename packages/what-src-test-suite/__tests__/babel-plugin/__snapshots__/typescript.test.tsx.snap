// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works as a babel-plugin with preset-typescript works in development mode 1`] = `
"var __WhatSrcGlobalVariable = require(\\"../../../node_modules/.cache/what-src\\");

import * as React from 'react';

class TestClassComponent extends React.Component {
  render() {
    return React.createElement(\\"div\\", {
      className: \\"class-component\\",
      \\"data-what-src\\": \\"1\\"
    }, React.createElement(\\"p\\", {
      \\"data-what-src\\": \\"2\\"
    }, \\"some p thing\\"));
  }

}

const TestFunctionalComponent = () => React.createElement(\\"div\\", {
  className: \\"functional-component\\",
  \\"data-what-src\\": \\"3\\"
}, React.createElement(\\"p\\", {
  \\"data-what-src\\": \\"4\\"
}, \\"some p thing\\"));

const TestMemoizedComponent = React.memo(() => React.createElement(\\"div\\", {
  className: \\"memoized-component\\",
  \\"data-what-src\\": \\"5\\"
}, React.createElement(\\"p\\", {
  \\"data-what-src\\": \\"6\\"
}, \\"some p thing\\")));"
`;

exports[`works as a babel-plugin with preset-typescript works in production mode 1`] = `
"var __WhatSrcGlobalVariable = require(\\"../../../node_modules/.cache/what-src\\");

import * as React from 'react';

class TestClassComponent extends React.Component {
  render() {
    return React.createElement(\\"div\\", {
      className: \\"class-component\\",
      \\"data-what-src\\": \\"1\\"
    }, React.createElement(\\"p\\", {
      \\"data-what-src\\": \\"2\\"
    }, \\"some p thing\\"));
  }

}

const TestFunctionalComponent = () => React.createElement(\\"div\\", {
  className: \\"functional-component\\",
  \\"data-what-src\\": \\"3\\"
}, React.createElement(\\"p\\", {
  \\"data-what-src\\": \\"4\\"
}, \\"some p thing\\"));

const TestMemoizedComponent = React.memo(() => React.createElement(\\"div\\", {
  className: \\"memoized-component\\",
  \\"data-what-src\\": \\"5\\"
}, React.createElement(\\"p\\", {
  \\"data-what-src\\": \\"6\\"
}, \\"some p thing\\")));"
`;
